texar/torch/modules/classifiers/bert_classifier.py

```diff
@@ -69,14 +69,14 @@ def __init__(self,
         super().__init__(hparams=hparams)
 
         # Create the underlying encoder
-        if self.__class__.__name__.startswith('BERT'):
+        if self._MODEL_NAME == 'BERT':
             encoder_hparams = dict_fetch(hparams, BERTEncoder.default_hparams())
 
             self._encoder = BERTEncoder(
                 pretrained_model_name=pretrained_model_name,
                 cache_dir=cache_dir,
                 hparams=encoder_hparams)
-        elif self.__class__.__name__.startswith('RoBERTa'):
+        elif self._MODEL_NAME == 'RoBERTa':
             encoder_hparams = dict_fetch(hparams,
                                          RoBERTaEncoder.default_hparams())
```

Comment:
Sorry but this is not what I meant. What I meant was something along the lines of:
```python
class BERTClassifier(...):
  _ENCODER_CLASS = BERTEncoder

  def __init__(...):
    ...
    encoder_hparams = dict_fetch(hparams, self._ENCODER_CLASS.default_hparams())
    ...

class RoBERTaClassifier(...):
  _ENCODER_CLASS = RoBERTaEncoder
```